src/DataTable/index.jsx

```diff
@@ -107,17 +108,30 @@ function DataTable({
   // Use the state and functions returned from useTable to build your UI
   const instance = useTable(...tableArgs);
 
-  // Call ``fetchData`` whenever the state of the table changes (e.g., page change, sort or filter applied) but ignore
-  // any state changes to current row selections as we don't want to re-fetch data whenever row(s) are selected.
-  const tableStateWithoutSelections = { ...instance.state };
-  delete tableStateWithoutSelections.selectedRowIds;
+  const {
+    pageSize: tableStatePageSize,
+    pageIndex: tableStatePageIndex,
+    sortBy: tableStateSortBy,
+    filters: tableStateFilters,
+    selectedRowIds: tableStateSelectedRowIds,
+  } = instance.state;
```

Comment:
These data are the same fields as were previously passed as arguments to `fetchData` below, so it is not a breaking change from that perspective. This change was necessary to prevent the `useEffect` that calls `fetchData` below from calling when selections were being made and allows us to move away from the more risky/hacky `delete` approach in the previous implementation.

Reply:
Also note, the original "controlled" selections implementation co-exists with this new approach. Need to sanity check that other uses of `DataTable` selections relying on the original selections implementations won't break as a result of this change.

Long term, we'll probably want to move away from or otherwise deprecate the current selections strategy, where "controlled" selections are not really controlled.